junifer/storage/hdf5.py

```diff
@@ -0,0 +1,954 @@
+            except IOError:
+                out_metadata = {}
+            # Update metadata
+            out_metadata.update(in_metadata)
+            # Save metadata
+            out_storage._write_processed_data(
+                fname=str(self.uri.resolve()),  # type: ignore
+                processed_data=out_metadata,
+                title="meta",
+            )
+            # Update element files for found MD5s
+            for feature_md5 in in_metadata.keys():
+                elements_per_feature_md5[feature_md5].append(file_)
+
+        # Run loop to collect data per feature per file
+        logger.info(
+            "Collecting data from "
+            f"{self.uri.parent}/*{self.uri.name}"  # type: ignore
+        )
+        for feature_md5, element_files in tqdm(
+            elements_per_feature_md5.items(), desc="feature"
+        ):
+            element_count = len(element_files)
+            # Chunk size for collecting
+            chunk_size = min(self.chunk_size, element_count)
+            # Operate on chunks
+            for chunk_idx, chunk_start in tqdm(
+                enumerate(range(0, element_count, chunk_size)), desc="chunk"
+            ):
+                # Store the chunk files' data
+                stored_data_for_chunk: List[Dict[str, Any]] = []
+                # Read the files of a chunk
+                for i in tqdm(
+                    range(chunk_start, chunk_start + chunk_size),
+                    desc="file-data",
+                ):
+                    file_ = element_files[i]
+                    logger.debug(
+                        f"Reading feature MD5: '{feature_md5}' "
+                        f"from HDF5 file: {file_} ..."
+                    )
+                    # Read from HDF5 and collect data
+                    stored_data_for_chunk.append(
+                        read_hdf5(
+                            fname=str(file_),
+                            title=feature_md5,
+                            slash="ignore",
+                        )
+                    )
+
+                # Concatenate the features data for a chunk
+                features_data = np.concatenate(
+                    [x["data"] for x in stored_data_for_chunk], axis=-1
+                )
+                # Make dictionary to write the collected data;
+                # first the static data then the dynamic data
+                data_to_write = {
+                    key: val
+                    for key, val in stored_data_for_chunk[0].items()
+                    if key not in ("data", "element")
+                }
+                # Join the features element for a chunk
+                data_to_write["element"] = reduce(
+                    lambda acc, elem: acc + elem,
+                    [x["element"] for x in stored_data_for_chunk],
+                    [],
+                )
+                # Write data in chunks to avoid memory usage spikes
+                if features_data.ndim == 2:
```

Comment:
you can avoid these two if clauses with one list and then convert to tuple.

Reply:
I know there's a bit of redundancy but kept it for explicit code. What do you mean exactly?

Reply:
The `ndim` would be 3 here but I get your point. Addressed in `07c769e4`.